src/Core/Usecase/HXL/Organisations/GetByUser.php

```diff
@@ -67,21 +67,26 @@ public function isSearch()
     public function interact()
     {
         // get user settings by user id
-        $user_settings = $this->userSettingRepository->getConfigKeyByUser($this->auth->getUserId(), 'hdx_api_key');
+        $user_settings_key = $this->userSettingRepository->getConfigKeyByUser($this->auth->getUserId(), 'hdx_api_key');
+        $user_settings_user_id = $this->userSettingRepository->getConfigKeyByUser(
+            $this->auth->getUserId(),
+            'hdx_user_id'
+        );
         // setup hdx interface
-        $this->setHDXInterface($user_settings);
+        $this->setHDXInterface($user_settings_key, $user_settings_user_id);
         $organisations = $this->hdxInterface->getAllOrganizationsForUser();
         if (!$organisations) {
             return $this->formatter->__invoke(null);
         }
         return $this->formatter->__invoke($organisations);
     }
 
-    private function setHDXInterface($user_settings)
+    private function setHDXInterface($user_settings_key, $user_settings_user_id)
     {
         $this->hdxInterface = new HDXInterface(
             getenv('HDX_URL'),
-            $user_settings->config_value
+            $user_settings_key->config_value,
```

Comment:
I think it might be better to restructure these values before passing them to `setHDXInterface` ie. only pass the config_value through, not the entire object. 
It's sort of weird that this method has to know about the config structure at all.

Reply:
That falls under refactor which is covered by a different ticket so I'm not going to change that here.